ImagePalette/UIImage+PaletteExtension.swift

```diff
@@ -50,26 +50,25 @@ extension UIImage {
         let bitmapData = malloc(bitmapByteCount)
         defer { free(bitmapData) }
 
-        if let context = CGBitmapContextCreate(bitmapData, pixelsWide, pixelsHigh, 8, bitmapBytesPerRow, colorSpace, CGImageAlphaInfo.PremultipliedFirst.rawValue) {
-            CGContextDrawImage(context, CGRectMake(0.0, 0.0, CGFloat(pixelsWide), CGFloat(pixelsHigh)), image)
+        if let context = CGContext(data: bitmapData, width: pixelsWide, height: pixelsHigh, bitsPerComponent: 8, bytesPerRow: bitmapBytesPerRow, space: colorSpace, bitmapInfo: CGImageAlphaInfo.premultipliedFirst.rawValue) {
+            context.draw(image, in: CGRect(x: 0.0, y: 0.0, width: CGFloat(pixelsWide), height: CGFloat(pixelsHigh)))
 
-            let unconstrainedData = CGBitmapContextGetData(context)
-            let data = UnsafePointer<UInt8>(unconstrainedData)
             var pixels = Array<Int64>()
-
-            for x in (0..<pixelsWide) {
-                for y in (0..<pixelsHigh) {
-                    let pixelInfo = ((pixelsWide * y) + x) * 4
-
-                    let alpha = Int64(data[pixelInfo])
-                    let red = Int64(data[pixelInfo + 1])
-                    let green = Int64(data[pixelInfo + 2])
-                    let blue = Int64(data[pixelInfo + 3])
-
-                    pixels.append(HexColor.fromARGB(alpha, red: red, green: green, blue: blue))
+            let unconstrainedData = context.data
+            if let data = UnsafePointer<UInt8>(unconstrainedData?.assumingMemoryBound(to: UInt8.IntegerLiteralType.self)) {
```

Comment:
That is the problem, I'd need Quizzes and its dependencies on Swift 3 right? That is the only thing this library is used for, to get a color of an image there, I think to match it w/ some design elements.